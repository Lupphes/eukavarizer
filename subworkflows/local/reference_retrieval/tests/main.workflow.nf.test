nextflow_workflow {

    name "Test Workflow REFERENCE_RETRIEVAL"
    script "../main.nf"
    workflow "REFERENCE_RETRIEVAL"
    config "subworkflows/local/reference_retrieval/tests/nextflow.config"
    tag "subworkflows"
    tag "subworkflows_local"
    tag "reference_retrieval"

    test("Download from RefSeq - stub") {
        options '-stub'

        when {
            params {
                outdir = "$outputDir"
            }
            workflow {
                """
                input[0] = 4932  // yeast taxonomy ID
                input[1] = "$outputDir"
                input[2] = []    // no user-provided reference
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.reference_genome != null },
                { assert workflow.out.reference_genome_unzipped != null },
                { assert workflow.out.reference_genome_bgzipped != null },
                { assert workflow.out.reference_genome_bwa_index != null },
                { assert workflow.out.reference_genome_faidx != null },
                { assert workflow.out.reference_genome_bgzipped_faidx != null },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("User-provided reference - stub") {
        options '-stub'

        when {
            params {
                outdir = "$outputDir"
            }
            workflow {
                """
                input[0] = 4932
                input[1] = "$outputDir"
                input[2] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.gz')
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.reference_genome != null },
                { assert workflow.out.reference_genome_unzipped != null },
                { assert workflow.out.reference_genome_bgzipped != null },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("BWA indexing path - stub") {
        options '-stub'

        when {
            params {
                outdir = "$outputDir"
                bwamem2 = false  // Use BWA not BWAMEM2
            }
            workflow {
                """
                input[0] = 4932
                input[1] = "$outputDir"
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.reference_genome_bwa_index != null }
            )
        }
    }

    test("BWAMEM2 indexing path - stub") {
        options '-stub'

        when {
            params {
                outdir = "$outputDir"
                bwamem2 = true  // Use BWAMEM2
            }
            workflow {
                """
                input[0] = 4932
                input[1] = "$outputDir"
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.reference_genome_bwa_index != null }
            )
        }
    }
}
