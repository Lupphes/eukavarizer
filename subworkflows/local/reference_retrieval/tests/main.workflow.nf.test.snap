{
    "Download from RefSeq - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "reference_genome": [
                    
                ],
                "reference_genome_bgzipped": [
                    
                ],
                "reference_genome_bgzipped_faidx": [
                    
                ],
                "reference_genome_bgzipped_index": [
                    
                ],
                "reference_genome_bwa_index": [
                    
                ],
                "reference_genome_faidx": [
                    
                ],
                "reference_genome_minimap_index": [
                    
                ],
                "reference_genome_ungapped_size": [
                    
                ],
                "reference_genome_unzipped": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-20T23:24:27.914728307"
    }
}